crates/wdl-engine/src/eval.rs

```diff
@@ -549,6 +549,13 @@ pub trait EvaluationContext: Send + Sync {
     /// Resolves a type name to a type.
     fn resolve_type_name(&self, name: &str, span: Span) -> Result<Type, Diagnostic>;
 
+    /// Gets the enums map for this evaluation context.
+    ///
+    /// Returns `None` if the context has not loaded enums.
+    fn enums(&self) -> Option<&HashMap<String, Enum>> {
```

Comment:
I think it would probably be best to lazy evaluate enum values instead of returning a map of names to "evaluated enums" (nor the need for the `Enum` type used here).

Perhaps a `fn resolve_variant_value(&self, name: &str, variant: &str) -> Result<Value, Diagnostic>` function instead (and called by the `value` stdlib function)?

Naively, this would look up the enum definition in the document associated with the evaluation context. From that, it would find the variant's value expression and evaluate it to produce a `Value`.

Ideally, we would cache that operation in a way that happens once per enum variant per document (the cache would need to be in the top-level evaluator).